kong/plugins/ai-proxy/handler.lua

```diff
@@ -33,6 +34,135 @@ local function internal_server_error(msg)
   return kong.response.exit(500, ERROR_MSG)
 end
 
+local function get_token_text(event_t)
+  -- chat
+  return
+    event_t and
+    event_t.choices and
+    #event_t.choices > 0 and
+    event_t.choices[1].delta and
+    event_t.choices[1].delta.content
+
+    or
+
+  -- completions
+    event_t and
+    event_t.choices and
+    #event_t.choices > 0 and
+    event_t.choices[1].text
+
+    or ""
+end
+
+local function handle_streaming_frame(conf)
+  -- make a re-usable framebuffer
+  local framebuffer = buffer.new()
+  local is_gzip = kong.response.get_header("Content-Encoding") == "gzip"
+
+  local ai_driver = require("kong.llm.drivers." .. conf.model.provider)
+
+  -- create a buffer to store each response token/frame, on first pass
+  if conf.logging
+       and conf.logging.log_payloads
+       and (not kong.ctx.plugin.ai_stream_log_buffer) then
+    kong.ctx.plugin.ai_stream_log_buffer = buffer.new()
+  end
+
+  -- now handle each chunk/frame
+  local chunk = ngx.arg[1]
+  local finished = ngx.arg[2]
+
+  if type(chunk) == "string" and chunk ~= "" then
+    -- transform each one into flat format, skipping transformer errors
+    -- because we have already 200 OK'd the client by now
+
+    if (not finished) and (is_gzip) then
```

Comment:
```suggestion
    if not finished and is_gzip then
```

Reply:
This doesn't work - it comes out as `if not (finished and is_gzip)` and broke the tests.